src/main/java/com/github/luben/zstd/util/Native.java

```diff
@@ -77,56 +77,59 @@ public static synchronized void load(final File tempFolder) {
                 throw linkError(e);
             }
         }
-        File tempLib;
+        File tempLib = null;
+        FileOutputStream out = null;
         try {
             tempLib = File.createTempFile(libname, "." + libExtension(), tempFolder);
+            // try to delete on exit, does not work on Windows
+            tempLib.deleteOnExit();
             // copy to tempLib
-            FileOutputStream out = new FileOutputStream(tempLib);
-            try {
-                byte[] buf = new byte[4096];
-                while (true) {
-                    int read = is.read(buf);
-                    if (read == -1) {
-                        break;
-                    }
-                    out.write(buf, 0, read);
+            out = new FileOutputStream(tempLib);
+            byte[] buf = new byte[4096];
+            while (true) {
+                int read = is.read(buf);
+                if (read == -1) {
+                    break;
                 }
+                out.write(buf, 0, read);
+            }
+            try {
+                out.flush();
+                out.close();
+                out = null;
+            } catch (IOException e) {
+                // ignore
+            }
+            try {
+                System.load(tempLib.getAbsolutePath());
+            } catch (UnsatisfiedLinkError e) {
+                // fall-back to loading the zstd-jni from the system library path
                 try {
-                    out.close();
-                    out = null;
-                } catch (IOException e) {
-                    // ignore
+                    System.loadLibrary(libnameShort);
+                } catch (UnsatisfiedLinkError e1) {
+                    // display error in case problem with loading from temp folder
+                    e.printStackTrace();
```

Comment:
We don't need this print here. Let the consumer decide what to do with the errors